reports-backend/src/main/java/org/jboss/da/reports/impl/ReportsGeneratorImpl.java

```diff
@@ -433,9 +434,36 @@ private BuiltReportModule toBuiltReportModule(GAV gav, VersionLookupResult vlr)
     @Override
     public List<LookupReport> getLookupReportsForGavs(LookupGAVsRequest request)
             throws CommunicationException{
+
+        /** Get set of GAs */
+        HashSet<GA> uniqueGAs = new HashSet<>();
+        request.getGavs().forEach((gav) -> uniqueGAs.add(gav.getGA()));
+
+        /** Get all the ProductArtifacts for the requested GAs */
+        Map<GA, CompletableFuture<Set<ProductArtifacts>>> gaProductArtifactsMap = getProductArtifactsPerGA(request, uniqueGAs);
+
+        /**  Use the mapping GA:Set<ProductArtifacts> and fill the LookupReports */
+        return createLookupReports(request, gaProductArtifactsMap);
+        
+    }
+
+    private Map<GA, CompletableFuture<Set<ProductArtifacts>>> getProductArtifactsPerGA(LookupGAVsRequest request, HashSet<GA> uniqueGAs) throws CommunicationException {
         Set<Product> products = getProducts(request.getProductNames(),
                 request.getProductVersionIds());
 
+        Map<GA, CompletableFuture<Set<ProductArtifacts>>> gaProductArtifactsMap = new HashMap<>();
+        uniqueGAs.stream()
+                .forEach((ga) -> {
```

Comment:
I don't know what is the benefit of stream here? Isn't regular forech more readable in this case? :wink: